### PR TITLE
Fix satisfaction survey link criteria

### PR DESCRIPTION
**Pull request by RomainLvr**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !30498

Different results between the satisfaction survey table in the personal view and the filter assigned to it in the view of tickets awaiting satisfaction survey.

The problem is a mistake between the two queries which are supposed to return the same result.

### CODE REVIEW

File: src/Ticket.php [hunk 1]
```diff
@@ -4971,17 +4971,21 @@ public static function showCentralList($start, $status = "process", bool $showgr
...
+                                'searchtype'  => 'equals',
+                                'value'       => Session::getLoginUserID(),
+                            ],
+                            [
+                                'link'        => 'OR',
+                                'field'       => 4, // requester
+                                'searchtype'  => 'equals',
+                                'value'       => Session::getLoginUserID(),
+                            ]
+                        ];
```
Comment (Rom1-B) [inline src/Ticket.php:4988]: The IF must remain
```suggestion
                        if (Session::haveRight('ticket', Ticket::SURVEY)) {
                            $options['criteria'][3]['link'] = 'AND';
                            $options['criteria'][3]['criteria'] = [
                                [
                                    'link'        => 'AND',
                                    'field'       => 22, // author
                                    'searchtype'  => 'equals',
                                    'value'       => Session::getLoginUserID(),
                                ],
                                [
                                    'link'        => 'OR',
                                    'field'       => 4, // requester
                                    'searchtype'  => 'equals',
                                    'value'       => Session::getLoginUserID(),
                                ]
                            ];
                        } else {
                            $options['criteria'][3]['field'] = 4; // requester
                            $options['criteria'][3]['searchtype'] = 'equals';
                            $options['criteria'][3]['value'] = Session::getLoginUserID();
                            $options['criteria'][3]['link'] = 'AND';
                        }
```

**Review by Rom1-B — CHANGES_REQUESTED** [review]

File: src/Ticket.php (hunk 1 above)
Comment (RomainLvr) [reply src/Ticket.php]: Aapplied, thanks :+1:

**Review by Rom1-B — APPROVED** [review]

**Review by trasher — APPROVED** [review]
Seems OK, need customer feedback

**Comment by RomainLvr** [comment]
> Seems OK, need customer feedback

The customer says he no longer has the error : [Ticket 30498](https://support.teclib.com/front/ticket.form.php?id=30498)

**Review by orthagh — APPROVED** [review]

**Pull request merged**